output/print_methods.go

```diff
@@ -2,59 +2,74 @@ package output
 
 import (
 	"fmt"
+	"net/http"
 	"strings"
 
 	"github.com/fatih/color"
 	"github.com/visola/go-http-cli/request"
 )
 
-type bodyBuffer struct {
-	*strings.Reader
-}
-
-func (bb *bodyBuffer) Close() error {
-	return nil
-}
-
 // PrintRequest outputs the http.Request
-func PrintRequest(request request.Request) error {
+func PrintRequest(request request.Request) {
 	boldGreen := color.New(color.Bold, color.FgGreen).PrintfFunc()
 	boldGreen("\n%s %s\n", request.Method, request.URL)
 
-	sentHeaderKeyColor := color.New(color.Bold, color.FgBlack).PrintfFunc()
-	sentHeaderValueColor := color.New(color.FgBlack).PrintfFunc()
+	printHeaders(request.Headers)
+	printCookies(request.Cookies)
+	printBody(request.Body, ">>")
+}
 
-	for headerName, values := range request.Headers {
-		sentHeaderKeyColor("%s:", headerName)
-		if len(values) > 1 {
-			for _, val := range values {
-				sentHeaderValueColor("\n  %s", val)
-			}
-			fmt.Println("")
-		} else {
-			sentHeaderValueColor(" %s\n", values[0])
+// PrintResponse outputs a http.Response
+func PrintResponse(response request.Response) {
+	if response.StatusCode < 300 {
```

Comment:
You might consider this overkill but after installing `github.com/jaebradley/go-http-status-class` (yes, it's my package 😉  ) you could do
```golang
sc, err := httpstatusclass.IdentifyStatusClass(response.StatusCode)

if err != nil {
  fmt.Printf("Unable to identify response status: %s\n", response.Status)
}

if sc == httpstatusclass.Informational || sc == httpstatusclass.Successful {
  color.Green("%s %s\n", response.Status, response.Protocol)
} else if sc == httpstatusclass.Redirection {
  color.Yellow("%s %s\n", response.Status, response.Protocol)
} else {
  color.Red("%s %s\n", response.Status, response.Protocol)
}
```
I find it slightly more readable, even though it _is_ more verbose.

Reply:
Yeah, I think this is overkill. As we discussed, there's always the impedance of reading and understanding what an added library does. But it's a nice idea.